### ISSUE TITLE
Magic Python regexp breaks a script

### ISSUE THREAD

**Issue by abbradar** [NixOS/nixpkgs#16133]
## Issue description

For [this script](https://github.com/eosrei/scfbuild/blob/master/bin/scfbuild) our magic regexp fails, resulting in this:
```
#!/nix/store/ng7iajdrlpa2v0wic6qdaknkrmqn9gli-python-2.7.11/bin/python2
# -*- coding: utf-8 -*-

from __future__ import (absolute_import, division, print_function,
import sys; sys.argv[0] = 'scfbuild'
                        unicode_literals)

import os
import sys
```

### Steps to reproduce

1. Build `pythonPackages.scfbuild` at revision 11b1291eb64500f13171b7b561e073b58246e413
2. Try to run the result.

**Comment by domenkozar** [comment]
I wish Python respected `exec -a`.

**Comment by domenkozar** [comment]
See also https://github.com/NixOS/nixpkgs/issues/11168

**Comment by abbradar** [comment]
I've fixed this in #17739.
